framework/src/materials/MaterialPropertyInterface.C

```diff
@@ -50,6 +52,8 @@ MaterialPropertyInterface::MaterialPropertyInterface(const MooseObject * moose_o
     _mi_tid(_mi_params.get<THREAD_ID>("_tid")),
     _stateful_allowed(true),
     _get_material_property_called(false),
+    _block_restricted(block_ids.find(Moose::ANY_BLOCK_ID) == block_ids.end()),
```

Comment:
This doesn't seem like the correct design. MaterialPropertyInterface shouldn't have these block/boundary restrictable Booleans as members. It's just an interface class. Can they not be moved to the TwoMaterialPropertyInterface class which is intended to understand block/boundary restriction? At the very least, you should be obtaining this information from the Block/Boundary restrictable objects directly, not recoding it here.